metric-anomaly-data-model/src/main/proto/org/hypertrace/alert/engine/eventcondition/config/service/v1/metric_anomaly_event_condition.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+
+package org.hypertrace.alert.engine.eventcondition.config.service.v1;
+
+message MetricAnomalyEventCondition {
+  MetricSelection metric_selection = 1;
+  repeated ViolationCondition violation_condition = 2;
+}
+
+message MetricSelection {
+  string key = 1;
+  string scope = 2;
+  MetricAggregationFunction metric_aggregation_function = 3;
+  string metric_aggregation_interval = 4; //iso duration string PT15S
+  repeated Filter filters = 5;
+}
+
+enum MetricAggregationFunction {
+  METRIC_AGGREGATION_FUNCTION_TYPE_UNSPECIFIED = 0;
+  METRIC_AGGREGATION_FUNCTION_TYPE_SUM = 1;
+  METRIC_AGGREGATION_FUNCTION_TYPE_AVG = 2;
+}
+
+message Filter {
+  oneof filter {
+    LeafFilter leaf_filter = 1;
+    CompositeFilter composite_filter = 2;
+  }
+}
+
+message LeafFilter {
+  LhsExpression lhs_expression = 1;
+  Operator operator = 2;
+  RhsExpression rhs_expression = 3;
+}
+
+enum Operator {
+  OPERATOR_UNSPECIFIED = 0;
+  OPERATOR_EQ = 1;
+  OPERATOR_AND = 2;
+  OPERATOR_OR = 3;
+}
+
+message CompositeFilter {
+  Operator operator = 1;
+  Filter lhs_filter = 2;
+  Filter rhs_filter = 3;
+}
+
+
+message LhsExpression {
+  oneof value {
+    ColumnIdentifier column_identifier = 1;
```

Comment:
the concept of column identifier leaks some implementation terminology that we can avoid (how does the consumer know what columns this uses?). If we want to avoid repeating scope here (I'm not sure that's necessary, since there's no real reason we can't mix scopes except convention and it makes the impl a bit more complicated since you have to join that information from the parent message), then we should rename it back to something like key so its clear its the same piece of data meant on line 13.

If it were me, I'd create an attribute message containing scope + key, and use that both here and in the selection object.

Reply:
Ok, let's do that. I just didn't want to repeat the scope again. But, we might need to mix scope in the future. So, go with the message attribute with key + scope. Done.